experiment/secret-sync-controller/main_test.go

```diff
@@ -1,295 +1,435 @@
+				Specs: []SecretSyncSpec{
+					{
+						Source: SecretManagerSpec{
+							Project: "k8s-jkns-gke-soak",
+							Secret:  "gsm-token",
+						},
+						Destination: KubernetesSpec{
+							Namespace: "missed",
+							Secret:    "secret-a",
+							Key:       "key-a",
+						},
+					},
 				},
-				Destination: KubernetesSpec{
-					Namespace: "ns1",
-					Secret:    "secret1",
-					Key:       "key1",
+			},
+
+			want: client.KubernetesSetup{
+				Namespaces: []client.NamespaceSetup{
+					{
+						Name: "ns-a",
+						Secrets: []client.K8sSecretSetup{
+							{
+								Name: "secret-a",
+								Keys: []client.KeySetup{
+									{
+										Name: "key-a",
+										Data: "old-token",
+									},
+								},
+							},
+						},
+					},
+					{
+						Name: "ns-b",
+						Secrets: []client.K8sSecretSetup{
+							{
+								Name: "secret-b",
+							},
+						},
+					},
+					{
+						Name: "ns-c",
+					},
 				},
 			},
-			want:      []byte{},
 			expectErr: true,
 		},
 		{
-			input: SecretSyncSpec{
-				Source: SecretManagerSpec{
-					Project: "projA",
-					Secret:  "secretA",
-				},
-				Destination: KubernetesSpec{
-					Namespace: "ns2",
-					Secret:    "secret3",
-					Key:       "key2",
+			// syncing from <non-existing gsm secret>.
+			// should return error
+			config: SecretSyncConfig{
+				Specs: []SecretSyncSpec{
+					{
+						Source: SecretManagerSpec{
+							Project: "k8s-jkns-gke-soak",
+							Secret:  "missed",
+						},
+						Destination: KubernetesSpec{
+							Namespace: "ns-a",
+							Secret:    "secret-a",
+							Key:       "key-a",
+						},
+					},
 				},
 			},
-			want:      []byte{},
-			expectErr: true,
-		},
-	}
-	for _, tt := range tests {
-		testname := fmt.Sprintf("%s", tt.input)
-		t.Run(testname, func(t *testing.T) {
-			client := clientMock{
-				K8sSecret: map[string]map[string]map[string][]byte{
-					"ns1": map[string]map[string][]byte{
-						"secret1": map[string][]byte{
-							"key1": []byte("valueOfSecret1"),
+			want: client.KubernetesSetup{
+				Namespaces: []client.NamespaceSetup{
+					{
+						Name: "ns-a",
+						Secrets: []client.K8sSecretSetup{
+							{
+								Name: "secret-a",
+								Keys: []client.KeySetup{
+									{
+										Name: "key-a",
+										Data: "old-token",
+									},
+								},
+							},
 						},
 					},
-					"ns2": map[string]map[string][]byte{
-						"secret2": map[string][]byte{
-							"key2": []byte("valueOfSecret2"),
+					{
+						Name: "ns-b",
+						Secrets: []client.K8sSecretSetup{
+							{
+								Name: "secret-b",
+							},
 						},
 					},
-				},
-				SecretManagerSecret: map[string]map[string][]byte{
-					"projA": map[string][]byte{
-						"secretA": []byte("valueOfSecretA"),
-					},
-					"projB": map[string][]byte{
-						"secretB": []byte("valueOfSecretB"),
+					{
+						Name: "ns-c",
 					},
 				},
+			},
+			expectErr: true,
+		},
+	}
+	for _, tt := range tests {
+		testname := fmt.Sprintf("%s", tt.config)
```

Comment:
consider adding a `name` field to your testcase table struct, then instead of commenting each test case you could name it

Reply:
Done.
Replaced.